src/machine/machine-varlink.c

```diff
@@ -825,3 +825,44 @@ int vl_method_bind_mount(sd_varlink *link, sd_json_variant *parameters, sd_varli
 
         return sd_varlink_reply(link, NULL);
 }
+
+int vl_method_open_root_directory_internal(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        _cleanup_close_ int fd = -EBADF;
+        Machine *machine = ASSERT_PTR(userdata);
+        Manager *manager = ASSERT_PTR(machine->manager);
+        int r;
+
+        r = sd_varlink_set_allow_fd_passing_output(link, true);
```

Comment:
i think instead of setting this lazily here i think it would be nicer to use the recently introduced SD_VARLINK_SERVER_ALLOW_FD_PASSING_OUTPUT flag when allocating the varlink server. See 6d4778f14f19a1787dce13641b4c6d84d9b715a6